injector_test.py

```diff
@@ -1583,6 +1584,15 @@ def test_binder_has_implicit_binding_for_implicitly_bound_type():
     assert not injector.binder.has_explicit_binding_for(int)
 
 
+def test_binder_with_uniqueness_checking_raises_error():
+    def configure(binder):
+        binder.bind(int, to=123)
+        binder.bind(int, to=456)
+
+    with pytest.raises(NonUniqueBinding):
+        _ = Injector([configure], unique=True)
```

Comment:
Good!

I would like to see how the `unique` flag interacts with parent/child injectors though, both to verify the functionality and as a form of documentation of the expected behavior. I would expect child injectors to be allowed to override binds in parent injectors (and I see no reason to why this should not work). What do you think?

Reply:
No problem! It looks like there are a couple of tests keying on similar concepts - I will borrow from them.